### ISSUE TITLE
t-pcie with sim7600 gps get latitude and longitude problem

### ISSUE THREAD

**Issue by tradexsrl** [Xinyuan-LilyGO/LilyGo-T-PCIE#22]
Hi , i was trying T-PCIE/examples/SIM7600/TinyGSM_Allfunctions/ examples with latest arduino ide and latest tinygsm library but i never get lat and lon.....
i setup gnss mode to galileo because i found by default gps module is configured for non european gps system but even let it run for 30 minutes the sketch i never get a signal lock from satellite
please let me know

**Comment by LilyGO** [comment]
Hello，What does the serial port print? Make sure the antenna is outdoors

**Comment by tradexsrl** [comment]
Hi , i was able to get lat lon after many hours of running the example but the result are negative..... for example lat -45 , lon -9

**Comment by LilyGO** [comment]
Are you not changing TinyGSM_Allfunctions?  Would you like to try this firmware? 
[PCIE_SIM7600X.zip](https://github.com/Xinyuan-LilyGO/LilyGo-T-PCIE/files/8108044/PCIE_SIM7600X.zip)

**Comment by tradexsrl** [comment]
hi , no changes on example tinygsm_all functions , i have sim7600E module , this firmware is for this one? because i see 7600X

**Comment by LilyGO** [comment]
7600E can use the firmware sent above.  Sim7600 cannot be upgraded in other modes.

**Comment by tradexsrl** [comment]
ok for the file , can i use this adapter to do the upgrade ?
https://techship.com/products/mpcie-to-usb-31-adapter/

**Comment by tradexsrl** [comment]
in the mean time can you test the 7600 simcom module to see with the example provided if you get lat and lon negative ?
i don't want the problem could be the tinygsm library

**Comment by LilyGO** [comment]
What should your GPS data be for the correct location?  The upgrade is just a matter of pulling out the USB cable

**Comment by tradexsrl** [comment]
correct latitude is 45 , correct longitude is 9 
i get latitude -45 , longitude -9

**Comment by LilyGO** [comment]
Use what firmware I sent you. Is the result the same?

**Comment by tradexsrl** [comment]
What version of firmware have you sent me? Because i have no way to do the upgrade but I'm able to get via at command the firmware I already had in the simcom module....maybe they are the same or maybe your fw is older than what I have in the simcom module ...and in this case the problem of getting negative lat Lon is still there

**Comment by tradexsrl** [comment]
this is the actual firmware: **SIMCOM_SIM7600E Revision: SIM7600M21-A_V2.0.1**
if the firmware you sent me is the same version , then is not usefull to try the upgrade and the problem still there

**Comment by LilyGO** [comment]
https://github.com/Xinyuan-LilyGO/LilyGo-T-PCIE/files/8108044/PCIE_SIM7600X.zip
This is for burning firmware for esp3
![1](https://user-images.githubusercontent.com/31160602/155840017-15c34572-70c0-4c97-8e84-17fcd812127e.png)
2

**Comment by tradexsrl** [comment]
tried with no luck...i have connected the t-pcie to my pc via usb cable with simcom7600 module mounted on top with simcard inserted with no pin, then i run the downloader 3.8.8 , setup com4 as the one recognized by my pc , then i setup all the things as in your screenshot but i get this one only :
![immagine](https://user-images.githubusercontent.com/62424179/155977283-cc6ba170-7106-435c-acc1-0250b9f150eb.png)

**Comment by tradexsrl** [comment]
and never get to finish in green as in your screenshot

**Comment by LilyGO** [comment]
Click "START" no normal burning? What about reopening the tool?

**Comment by tradexsrl** [comment]
i click on start....and i get stop in brown....i tried again now , same problem.
i suspect the tools needs a way to go through esp32 and get to simcom7600 module via a softwareserial...so you need a sketch to load to esp32 and enable this passthrough

**Comment by LilyGO** [comment]
This is the firmware for the ESP32. Connect type-c directly

**Comment by tradexsrl** [comment]
i don't understand...the connector type-c on the t-pcie board is connected to my pc...there is no any other connector on the board...

**Comment by LilyGO** [comment]
No ports?  What kind of computer do you have, with a drive?  Does the board start normally?

**Comment by tradexsrl** [comment]
hi , i think we don't understand each other....simcom7600 is mounted on t-pcie board and this is connected to my pc but with download tool i have the problem written...see the picture
![immagine](https://user-images.githubusercontent.com/62424179/156520591-7b789a30-688e-4e8b-b012-070fbae9116a.png)

**Comment by LilyGO** [comment]
You are right to connect. Is there a port? Can it burn normally?

**Comment by tradexsrl** [comment]
yes in my pc there is com4 and i can program normally with arduino IDE but NOT with the tool suggested by you to upgrade the simcom firmware.
let me know once and for all with the correct and detailed procedure to do that.
